### ISSUE TITLE
Enhancement: Add a [consider-using-get]-checker

### ISSUE THREAD

**Issue by pheanex** [PyCQA/pylint#2076]
**I suggest adding a checker for unidiomatic "getting" of default values from a missing dictionary-key**

Example:
```python
# bad
dictionary = dict()
if 'key' in dictionary:
    variable = dictionary['key']
else:
    variable = 'default'

# good
variable = dictionary.get('key', 'default')
```
The idiomatic way would be to use the dict.get method with a default "if missing" value.
See i.e. [docs.python-guide.org](http://docs.python-guide.org/en/latest/writing/style/#access-a-dictionary-element) and [docs.python.org](https://docs.python.org/3/library/stdtypes.html#dict.get)

For a first set of test-cases I came up with the following (feel free to suggest additions):

```python
dictionary = dict()

if 'key' in dictionary:  # [consider-using-get]
    variable = dictionary['key']
else:
    variable = 'default'

if 'key' in dictionary:  # [consider-using-get]
    variable = dictionary['key']


if 'key1' in dictionary:  # not the simple case
    variable = dictionary['key1']
elif 'key2' in dictionary:
    variable = dictionary['key2']
else:
    variable = 'default'

if 'key' in dictionary:  # body is not pure
    variable = dictionary['key']
    print('found')

if 'key' in dictionary:  # body is not pure
    variable = dictionary['key']
    print('found')
else:
    variable = 'default'

if 'key' in dictionary:  # body is not pure
    variable = dictionary['key']
else:
    variable = 'default'
    print('found')

if 'key1' in dictionary:  # different keys
    variable = dictionary['key2']
else:
    variable = 'default'

if 'key' in dictionary1:  # different dictionaries
    variable = dictionary2['key']
else:
    variable = 'default'

if 'key' in dictionary:  # different variables
    variable1 = dictionary['key']
else:
    variable2 = 'default'

if 'key' in dictionary:  # assignment is not simple
    variable1 = variable2 = dictionary['key']

if 'word' in 'text':
    variable = 'text'['word']  # already bogus, but to assert that this only works with dictionaries
```

For the implementation of the checker I was thinking of looking for the specific structure of the if/else construct and using inferrence for checking if the checked against variable is a dictionary type.

How do you feel about this enhancement?
Thoughs, comments, objections?

**Comment by PCManticore** [comment]
This could be a good enhancement, thanks for the suggestion!
